### PR TITLE
test(coverage): B10 — cli/commands to ≥80% branch (13 files, 70→57)

### PR DESCRIPTION
**Pull request by asafgolombek**
## Summary

Sub-project **B10** of the True Coverage Program (close per-subsystem branch gaps) — the first **non-connector** slice. Raises all 13 below-floor `cli/commands` files above the ≥80% line+branch floor.

Files cleared (main branch% before): `doctor-core` (67.72, line 77.91), `config` (70.49), `bench` (72.73), `query` (73.96), `audit` (74.19), `identity` (75), `run-workflow` (75, line 78), `extension` (75.1), `scim` (76), `status` (76.39), `update` (77.27), `deploy` (77.94), `people` (78.31).

## Approach

- Extended each command's existing sibling test via the established cli **DI harness**: `createMockIpcClient` (injected fake IPC client), `captureOutput`, and `setFixture`/`FAKE_SOCKET_PATH` — driving both the arg-parsing layer (`run<Name>`) and the logic layer (`run<Name>Impl`).
- **No source changes, no new DI seams** — every command already exposed the necessary seam.
- **No `any`, no `mock.module`, no `biome-ignore`.** `mock.module` is deliberately avoided (it contaminates the combined `bun test packages/cli/src` run on CI); the full combined cli suite (1394 tests) is green.
- process.exitCode reset per-test; now-relative date/version fixtures.

## Baseline

- `coverage-baseline.json` 70 → **57** — pure 13-file removal, 0 added, 0 watermark drift on retained files.

Reseed: Docker `oven/bun:latest` (bun 1.3.14 = CI, Linux-authoritative).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Expanded CLI test coverage across many commands to exercise edge cases, error handling, output/formatting branches, dispatcher wiring, stdin/TTY behaviors, and exit-code mappings.
* **Chores**
  * Updated coverage baseline generation timestamp.

### CODE REVIEW

File: packages/cli/src/commands/config.test.ts [hunk 1]
```diff
@@ -207,14 +272,51 @@ describe("runConfigEdit", () => {
...
+      await runConfigEdit("/tmp/nimbus.toml", spawnStub);
+    } finally {
+      if (savedEditor === undefined) {
+        delete process.env["EDITOR"];
+      } else {
+        process.env["EDITOR"] = savedEditor;
+      }
+    }
+    expect(calls[0]?.cmd).toBe("myeditor");
+  });
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/cli/src/commands/config.test.ts:302]: Fixed in 0bac4a4d — replaced `"/tmp/nimbus.toml"` with `join(makeTmp(), "nimbus.toml")`.

File: packages/cli/src/commands/doctor-core.test.ts [hunk 2]
```diff
@@ -470,3 +470,241 @@ describe("doctorVoiceLines", () => {
...
+        readGatewayState: async () => undefined,
+      }),
+    );
+
+    // Voice lines with no binaries produce [warn] messages that escalate exit.
+    // The gateway is also absent (exit=2 from "[fail] Gateway: not running").
+    expect(out.stdout).toContain("[fail] Gateway: not running");
+    // Exit is at least 2 (from gateway check)
+    expect(process.exitCode).toBeGreaterThanOrEqual(1);
+  });
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/cli/src/commands/doctor-core.test.ts:623]: Fixed in 0bac4a4d — the test now also asserts a concrete `[warn] Voice:` line (`expect(out.stdout).toMatch(/\[warn\] Voice:/)`), so the voice branch is proven to run rather than relying on the exit code (which the absent gateway already escalates).

File: packages/cli/src/commands/identity.test.ts [hunk 3]
```diff
@@ -229,3 +232,180 @@ describe("runIdentityCommand — login (full flow via runIdentityCommand)", () =
...
+      exitedWith = code;
+      throw new Error(`process.exit(${code})`);
+    }) as typeof process.exit;
+    try {
+      await runIdentity(["bogus-sub"]).catch(() => {});
+    } finally {
+      process.exit = origExit;
+    }
+    expect(exitedWith).toBe(1);
+  });
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/cli/src/commands/identity.test.ts:381]: Fixed in 0bac4a4d. `runIdentity` parses argv *before* reading gateway state, so the unknown-subcommand parse error fires regardless of gateway presence. The error is written via `process.stderr.write` (not `console.error`, which `captureOutput` captures), so — mirroring the gateway-not-running test — the test now patches `process.stderr.write` (restored in `finally`) and asserts it saw `"Unknown subcommand"`, proving the parse branch (not the gateway path) ran.

File: packages/cli/src/commands/query.test.ts [hunk 4]
```diff
@@ -189,3 +199,302 @@ describe("runQuery — --service path", () => {
...
+        meta: { limit: 50, total: 1 },
+      },
+    ]);
+    await runQuery(["--service", "svc", "--pretty"]);
+    expect(out.stdout).toContain("same");
+    // body_preview === title → not printed twice; only once via title line
+    const occurrences = out.stdout.split("same").length - 1;
+    // printed at most twice (title line + footer), NOT three times
+    expect(occurrences).toBeLessThan(3);
+  });
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/cli/src/commands/query.test.ts:478]: Fixed in 0bac4a4d — tightened to `expect(occurrences).toBe(1)`. When `body_preview === title` the body line is suppressed, so `"same"` renders exactly once (the title line); if it weren't suppressed it would appear twice. The old `< 3` passed even on a double-print.

File: packages/cli/src/commands/run-workflow.test.ts [hunk 5]
```diff
@@ -157,21 +157,251 @@ describe("runWorkflowFromFileWithClient", () => {
...
+      dryRun: true,
+      noTtv: true,
+      agent: undefined,
+    });
+    expect(ipc.calls).toHaveLength(2);
+    expect(ipc.calls[0]?.method).toBe("workflow.save");
+    expect(ipc.calls[1]?.method).toBe("workflow.run");
+    expect(ipc.calls[1]?.params).toMatchObject({ dryRun: true, stream: false });
+  });
 });
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/cli/src/commands/run-workflow.test.ts:429]: Fixed in 0bac4a4d — added a HITL-gate negative case: a `--no-ttv` preview whose `stepResults` contain `hitlActions` must `throw` and NEVER issue the real `workflow.run`. The test asserts `.rejects.toThrow(/human approval \(HITL\)/)` and that exactly two IPC calls ran (`workflow.save` + the preview `workflow.run`) — proving the gate stops before the real run.

File: packages/cli/src/commands/audit.test.ts [hunk 6]
```diff
@@ -194,4 +227,73 @@ describe("runAudit (dispatcher)", () => {
...
+    };
+    try {
+      await runAudit(["export", "--output", outPath]);
+    } finally {
+      (Bun as { write: unknown }).write = origWrite;
+    }
+    expect(ipc.calls[0]).toEqual({ method: "audit.exportAll", params: {} });
+    expect(writes[0]?.path).toBe(outPath);
+    expect(out.stdout).toContain(`wrote 1 audit rows to ${outPath}`);
+  });
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/cli/src/commands/audit.test.ts:278]: The `Bun.write` patch (audit) and `process.stdin` patch (update) are each saved and restored in a `try/finally` within the test, so they're process-local and don't contaminate the combined run — verified: `bun test packages/cli/src` is green on Linux (oven/bun:latest, CI=true), 1394 pass. (The separate cross-file leak this PR fixed was a non-restored, non-writable `isTTY` redefinition, a different mechanism.) Adding injectable seams at the `runAudit()`/`runUpdate()` entrypoints is a reasonable refactor but is a source change beyond this coverage slice; keeping the restored-patch approach.

File: packages/cli/src/commands/config.test.ts [hunk 7]
```diff
@@ -254,4 +356,61 @@ describe("runConfig (dispatcher)", () => {
...
+      calls.push(cmd);
+      const emitter = new EventEmitter();
+      queueMicrotask(() => {
+        emitter.emit("close", 0);
+      });
+      return emitter;
+    };
+    await runConfigEdit(join(tmp, "nimbus.toml"), spawnStub);
+    expect(calls).toHaveLength(1);
+  });
```
[the remark this answers is not in the document]
Comment (asafgolombek) [reply packages/cli/src/commands/config.test.ts:415]: `runConfig()` derives `tomlPath` from `getCliPlatformPaths()`, and the `edit` branch spawns a real editor — neither is injectable at the dispatcher layer (only `runConfigEdit` exposes a `spawnFn` seam). The dispatcher **routing** for these subcommands is already proven by the tests that go through `runConfig(["get"])` / `runConfig(["set"])` and reach the helper's usage-error guard. The happy-path helpers are exercised directly because routing them through `runConfig()` would write to the real platform config dir / spawn a real editor. Adding `getCliPlatformPaths`/`spawnFn` DI to `runConfig` is a reasonable future improvement but out of scope for this coverage PR.
